src/Algebra/Ordered/Structures.agda

```diff
@@ -166,6 +166,43 @@ record IsProsemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂)
 
   open IsSemiring isSemiring public using (distribˡ; distribʳ; zeroˡ; zeroʳ)
 
+-- Preordered IdempotentSemiring (IdempotentProsemiring)
+
+record IsIdempotentProsemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+  field
+    isProsemiring  : IsProsemiring + * 0# 1#
+    +-idem         : Idempotent +
+
+  open IsProsemiring isProsemiring public
+
+  isIdempotentSemiring : IsIdempotentSemiring + * 0# 1#
+  isIdempotentSemiring = record { isSemiring = isSemiring ; +-idem = +-idem }
+
+  open IsIdempotentSemiring isIdempotentSemiring public using (+-idem)
+
+-- Preordered KleeneAlgebra (prokleenealgebra)
+
+record IsProkleenealgebra (+ * : Op₂ A) ( ⁻* : Op₁ A) (0# 1# : A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
```

Comment:
I think it should definitely be `IsProKleeneAlgebra` no? Similarly with the other names....

Reply:
Not Sure I prefer `IsProKleeneAlgebra` but looking at other names it should be `IsProkleeneAlgebra`? I will change it to `IsProKleeneAlgebra` and `IsPoKleeneAlgebra`

Reply:
Thanks, `IsProKleeneAlgebra` and `IsPoKleeneAlgebra` look better.